homeassistant/helpers/entity_component.py

```diff
@@ -230,12 +231,30 @@ def async_register_entity_service(
         if isinstance(schema, dict):
             schema = cv.make_entity_service_schema(schema)
 
-        async def handle_service(call: ServiceCall) -> ServiceResponse:
+        async def handle_service(
+            call: ServiceCall,
+        ) -> EntityServiceResponse | ServiceResponse:
             """Handle the service."""
-            return await service.entity_service_call(
+
+            result = await service.entity_service_call(
                 self.hass, self._platforms.values(), func, call, required_features
             )
 
+            if supports_response == SupportsResponse.ONLY_LEGACY:
+                self.logger.warning(
```

Comment:
I think we should (1) move this into the service that is being deprecated and (2) add this only when we have the replacement service ready

Reply:
Will add it in a follow up PR for calendar and weather

Reply:
https://github.com/home-assistant/core/pull/102481